File: php/src/vaas/Message/VerdictResponse.php
<?php

namespace VaasSdk\Message;

class VerdictResponse extends BaseMessage
{
    public Verdict $verdict;
    public ?string $url;
    public string $guid;
    public string $sha256;
    public ?string $upload_token;
    public ?string $file_type;
    public ?string $mime_type;
    public ?string $detection;
}

File: golang/vaas/pkg/messages/verdict_response.go
// Package messages provides structures for handling communication messages between the client and the VaaS server.
package messages

// VerdictResponse represents the response containing the verdict information.
type VerdictResponse struct {
	Kind        string  `json:"kind"`
	Sha256      string  `json:"sha256"`
	GUID        string  `json:"guid"`
	Verdict     Verdict `json:"verdict"`
	URL         string  `json:"url"`
	UploadToken string  `json:"upload_token"`
	Detection   string  `json:"detection"`
	FileType    string  `json:"file_type"`
	MimeType    string  `json:"mime_type"`
}

// IsValid checks if the VerdictResponse is valid.
func (response VerdictResponse) IsValid() bool {
	return response.Sha256 != "" && response.Verdict != ""
}
